src/main/scala/com/gradle/internal/CustomBuildScanEnhancements.scala

```diff
@@ -1,328 +1,370 @@
-
-    if (isNotEmpty(gitRepo)) {
-      buildScan.addValue("Git repository", redactUserInfo(gitRepo.get))
-    }
-    if (isNotEmpty(gitCommitId)) {
-      buildScan.addValue("Git commit id", gitCommitId.get)
-    }
-    if (isNotEmpty(gitCommitShortId)) {
-      // Ensure server URL is configured by deferring call at execution time
-      addCustomValueAndSearchLink("Git commit id", "Git commit id short", gitCommitShortId.get)
-    }
-    if (isNotEmpty(gitBranchName)) {
-      buildScan.tag(gitBranchName.get)
-      buildScan.addValue("Git branch", gitBranchName.get)
-    }
-    if (isNotEmpty(gitStatus)) {
-      buildScan.tag("Dirty")
-      buildScan.addValue("Git status", gitStatus.get)
+  private val captureBuildkite: BuildScan => BuildScan = {
+    if (!CiUtils.isBuildkite) identity
+    else {
+      val prSource = for {
+        repository <- env.envVariable[String]("BUILDKITE_PULL_REQUEST_REPO")
+        webRepoUri <- Utils.toWebRepoUri(repository)
+        prNumber <- env.envVariable[String]("BUILDKITE_PULL_REQUEST")
+      } yield sbt.url(s"$webRepoUri/pull/$prNumber")
+
+      val ops = Seq(
+        (bs: BuildScan) => bs.value("CI provider", "Buildkite"),
+        ifDefined(env.envVariable[URL]("BUILDKITE_BUILD_URL"))(_.link("Buildkite build", _)),
+        ifDefined(env.envVariable[String]("BUILDKITE_COMMAND"))(withCustomValueAndSearchLink(_, "CI command", _)),
+        ifDefined(env.envVariable[String]("BUILDKITE_BUILD_ID"))(_.value("CI build number", _)),
+        ifDefined(prSource)(_.link("PR source", _))
+      )
+      Function.chain(ops)
     }
+  }
 
-    val gitHubUrl = envVariable("GITHUB_SERVER_URL")
-    val gitRepository = envVariable("GITHUB_REPOSITORY")
-    if (gitHubUrl.isDefined && gitRepository.isDefined && isNotEmpty(gitCommitId)) {
-      buildScan.link("GitHub source", gitHubUrl.get + "/" + gitRepository.get + "/tree/" + gitCommitId.get)
-    }
+  private val captureCiMetadata: BuildScan => BuildScan = {
+    val ops = Seq(
+      captureJenkinsOrHudson,
+      captureTeamCity,
+      captureCircleCi,
+      captureBamboo,
+      captureGitHubActions,
+      captureGitLab,
+      captureTravis,
+      captureBitrise,
+      captureGoCd,
+      captureAzurePipelines,
+      captureBuildkite
+    )
+    Function.chain(ops)
+  }
+
+  private val captureGitMetadata: BuildScan => BuildScan = {
+    if (!isGitInstalled) identity
     else {
-      if (isNotEmpty(gitRepo) && isNotEmpty(gitCommitId)) {
-        val webRepoUri = toWebRepoUri(gitRepo.get)
-        webRepoUri.foreach(uri => {
-          if (uri.getHost.contains("github")) {
-            buildScan.link("GitHub source", uri + "/tree/" + gitCommitId.get)
-          }
-          else {
-            if (uri.getHost.contains("gitlab")) {
-              buildScan.link("GitLab source", uri + "/-/commit/" + gitCommitId.get)
-            }
-          }
-        })
+      val gitRepo = Utils.execAndGetStdOut("git", "config", "--get", "remote.origin.url")
```

Comment:
Can we use static imports to keep consistency with the Java version here and in all the other places in this file that previously used them?